torch/testing/_internal/common_distributed.py

```diff
@@ -111,6 +114,8 @@ def wrapper(*args, **kwargs):
         world_size = int(os.environ["WORLD_SIZE"])
         if torch.cuda.device_count() < world_size:
             sys.exit(TEST_SKIPS[f"multi-gpu-{world_size}"].exit_code)
+        if TEST_HPU and torch.hpu.device_count < world_size:
+            sys.exit(TEST_SKIPS[f"multi-gpu-{world_size}"].exit_code)
```

Comment:
Can we use the `device_handle` way (as later in this PR) for every device type?

We can have a global `device_handle` setter that set it based on `TEST_HPU` or something else.